app/controllers/orders_controller.rb

```diff
@@ -9,24 +9,28 @@ def show
   end
 
   def create
-    order = Order.create(order_params)
+    order = current_user.orders.new(order_params)
     if order.save
       cart.items.each do |item,quantity|
-        order.item_orders.create({
+        order.item_orders.create!({
           item: item,
           quantity: quantity,
           price: item.price
           })
       end
       session.delete(:cart)
-      redirect_to "/orders/#{order.id}"
+      if current_user.role == "admin"
```

Comment:
23 and 24 are ready to be deleted!

Reply:
Updated by @LHJE - actually likes 22-23 in our atom